### PR TITLE
Update hardware IDs to match current assignments in OANA.

### PR DESCRIPTION
**Pull request by luqmana**
Update propolis to reflect the current set of IDs assigned in [OANA (Oxide Assigned Numbers Authority)](https://github.com/oxidecomputer/oana).

### CODE REVIEW

File: propolis/src/hw/virtio/pci.rs [hunk 1]
```diff
@@ -186,10 +185,10 @@ impl PciVirtioState {
         cfg_sz: usize,
     ) -> (Self, pci::DeviceState) {
         let mut builder = pci::Builder::new(pci::Ident {
-            vendor_id: VIRTIO_VENDOR,
+            vendor_id: VENDOR_VIRTIO,
             device_id: dev_id,
-            sub_vendor_id: VIRTIO_VENDOR,
-            sub_device_id: dev_id - 0xfff,
+            sub_vendor_id: VENDOR_OXIDE,
+            sub_device_id: super::bits::virtio_sub_dev_id(dev_id),
```
Comment (pfmooney) [inline propolis/src/hw/virtio/pci.rs:191]: These could probably just be passed through as parameters from the specific device?  We could then make them configurable later, if it was required for something like testing.

File: propolis/src/hw/nvme/mod.rs [hunk 2]
```diff
@@ -422,16 +424,14 @@ pub struct PciNvme {
 impl PciNvme {
     /// Create a new pci-nvme device with the given values
     pub fn create(
-        vendor: u16,
-        device: u16,
```
Comment (pfmooney) [inline propolis/src/hw/nvme/mod.rs:426]: Why not keep this stuff as a parameter?

Comment (luqmana) [reply propolis/src/hw/nvme/mod.rs:426]: NVMe was the only device that took them as parameters so I was just matching everywhere else.

File: propolis/src/hw/virtio/pci.rs (hunk 1 above)
Comment (luqmana) [reply propolis/src/hw/virtio/pci.rs:191]: Fair enough, but we don't have any such use cases today? What cases do you see it being useful to configure the IDs in a test?

**Review by pfmooney — APPROVED** [review]

**Pull request merged**